packages/valory/skills/decision_making_abci/tests/test_task_preparations.py

```diff
@@ -105,11 +105,15 @@ def test_check_conditions_task_disabled(
         """Test check_conditions when the task is disabled."""
 
         self.mock_task_preparation.enabled = False
-        for value in self.mock_task_preparation.check_conditions():
-            assert not value
-            self.mock_task_preparation.logger.info.assert_called_with(
-                f"[{self.__class__.__name__}]: task is disabled"
-            )
+        gen = self.mock_task_preparation.check_conditions()
+
+        with pytest.raises(StopIteration) as excinfo:
+            next(gen)
+        exception_message = False
+        assert str(exception_message) in str(excinfo.value)
+        self.mock_task_preparation.logger.info.assert_called_with(
+            "[TaskPreparation]: task is disabled"
+        )
```

Comment:
Lots of code dup, it's the same until line 188. How about a parametrized test?

Reply:
still lots of code dup but no time to refactor this into the parameterised test atm. Hopefully I'll get some time in the future to do that.